docs/source/libraries/xgboost.rst

```diff
@@ -5,15 +5,21 @@ XGBoost
 
 XGBoost_ is a popular Gradient Boosting library with Python interface.
 eli5 supports :func:`eli5.explain_weights` and :func:`eli5.explain_prediction`
-for XGBClassifer_ and XGBRegressor_ estimators. It is tested for
+for XGBClassifer_, XGBRegressor_ and Booster_ estimators. It is tested for
 xgboost >= 0.6a2.
 
 .. _XGBoost: https://github.com/dmlc/xgboost
 .. _XGBClassifer: https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBClassifier
 .. _XGBRegressor: https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBRegressor
+.. _Booster: http://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.Booster
 
-:func:`eli5.explain_weights` uses feature importances. Additional
-arguments for XGBClassifer_ and XGBRegressor_:
+:func:`eli5.explain_weights` uses feature importances.
+There is one additional required argument for Booster_ estimator:
+
+* ``is_regression`` True if solving a regression problem
```

Comment:
For explaining xgboost weights, is_regression does not matter at all, because only importances are shown and they don't include the class/target label:
![image](https://cloud.githubusercontent.com/assets/424613/26552127/7d7d64b2-448e-11e7-83fc-6eb1a7542923.png)
For explain_prediction, the difference is this (is_regression=False) 
![image](https://cloud.githubusercontent.com/assets/424613/26552231/eef6ca02-448e-11e7-9c18-f22b9a362980.png)

vs. this (is_regression=True)

![image](https://cloud.githubusercontent.com/assets/424613/26552242/fcf541ba-448e-11e7-99f7-1a142c5e2533.png)

For xgboost, we get proba from actual predictionis, and score from calculation over tree dump - they are equal in case of regression.

Still thinking about this, but I agree that it would be better to remove the is_regression argument or at least make it optional.